### PR TITLE
MAINT: change the default NN for ANN-based regionalization

### PR DESCRIPTION
**Pull request by nghi-truyen**
Enhance the flexibility of the default NN in various spatial domain dimensions:
- The previous number of neurons in the hidden layer count `sqrt(nd * x * y) * 2 / 3` caused excessive parameter growth as the spatial domain increased. The new formulation `sqrt(nd * ncv) * log(x * y)` that applies a logarithmic function to x and y could address this issue.
- Hyperbolic tangent function is used as the output layer instead of Sigmoid.

### CODE REVIEW

File: doc/source/user_guide/classical_uses/lez_regionalization.rst [hunk 1]
```diff
@@ -269,8 +269,8 @@ Since we have returned the optimized neural network, we can visualize what it co
 
     opt_ann.net
 
-The information displayed tells us that the default neural network is composed of 2 hidden dense layers followed by ``ReLU`` activation functions 
-and a final layer followed by a ``Sigmoid`` function. To scale the network output to the boundary condition, a ``MinMaxScale`` function is applied. 
+The above information indicates that the default neural network is composed of 3 hidden dense layers, each followed by a ``ReLU`` activation function.
+The output layer is followed by a ``TanH`` function. To scale the network output to the boundary condition, a ``MinMaxScale`` function is applied.
```
Comment (pag13) [inline doc/source/user_guide/classical_uses/lez_regionalization.rst:273]: The output layer is followed by a "tanh" hyperbolic tangent function and it outputs in \left]-1;1\right[ are scaled  to given conceptual parameter bounds using a ``MinMaxScale`` function.

File: smash/core/simulation/_standardize.py [hunk 2]
```diff
@@ -330,63 +330,65 @@ def _standardize_simulation_optimize_options_descriptor(
...
-        # % check bounds constraints
-        if hasattr(net.layers[-1], "_scale_func"):
-            net_bounds = np.transpose([net.layers[-1]._scale_func.lower, net.layers[-1]._scale_func.upper])
+                diff = np.not_equal(net_bounds, bound_values)
 
-            diff = np.not_equal(net_bounds, bound_values)
+                for i, name in enumerate(bounds):
+                    if diff[i].any():
+                        warnings.warn(
+                            f"net optimize_options: Inconsistent values between the bound in scaling layer "
```
Comment (pag13) [inline smash/core/simulation/_standardize.py:381]: between the bounds

**Review by pag13 — APPROVED** [review]
Ok for this new formulation for the hidden layer size; for tanh instead of sigmoid in output layer.

File: doc/source/user_guide/classical_uses/lez_regionalization.rst (hunk 1 above)
Comment (nghi-truyen) [reply doc/source/user_guide/classical_uses/lez_regionalization.rst:273]: Exactly !

File: smash/core/simulation/_standardize.py [hunk 3]
```diff
@@ -330,63 +330,65 @@ def _standardize_simulation_optimize_options_descriptor(
 def _standardize_simulation_optimize_options_net(
     model: Model, bounds: dict, net: Net | None, **kwargs
 ) -> Net:
-    nd = model.setup.nd
+    x, y, nd = model.physio_data.descriptor.shape
```
Comment (inoelloc) [inline smash/core/simulation/_standardize.py:333]: Je suis pas hyper fan de `x, y`. Ca porte à confusion avec ce que ca represente vraiment. Ici, `x = nrow` et `y = ncol`.
Soit inverser l'ordre de `x, y`:
``` python
 y, x, nd = model.physio_data.descriptor.shape
```
Soit utiliser `nrow, ncol`:
``` python
 nrow, ncol, nd = model.physio_data.descriptor.shape
```
Je pref solution 2 ;)

Comment (nghi-truyen) [reply smash/core/simulation/_standardize.py:333]: yess

**Pull request merged**